ca/server.go

```diff
@@ -521,19 +524,21 @@ func (s *Server) isRunning() bool {
 // anything to update the root CA material
 func (s *Server) UpdateRootCA(ctx context.Context, cluster *api.Cluster) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
Wondering if any harm can come from holding the lock for the whole function. I'm not sure whether we weren't before as an optimization, or because there's a long-lived operation here, or one that might deadlock.